### PR TITLE
meson: Use `configuration_data()` instead of piping variables through `echo`

### PR DESCRIPTION
**Pull request by MarijnS95**
Having a template file with replacement macros is a bit cleaner than encoding the setup of the file in `meson.build` and piping that through `echo` with caputred variables.

Unfortunately there appears to be no support for repetition, to e.g. expand a list of platform names into the various uses (array with comma-separation, or individual `extern struct` declarations, forcing us to have some form of string expansion within `meson.build`).

### CODE REVIEW

**Comment by lumag** [comment]
I think `platforms.c.in` is missing

**Comment by MarijnS95** [comment]
> I think `platforms.c.in` is missing

Testing whether the CI could catch that :smirk:

File: platforms.c.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* Autogenerated file, do not edit */
+#include <stdlib.h>
+
+@PLATFORM_DEFS@
+const struct debugcc_platform *platforms[] = {
+@PLATFORM_ARRAY@	NULL,
```
Comment (lumag) [inline platforms.c.in:6]: Could you please add a newline before NULL?

**Review by lumag — CHANGES_REQUESTED** [review]

File: platforms.c.in (hunk 1 above)
Comment (MarijnS95) [reply platforms.c.in]: Was just trying out whether to use `'\n'.join()` which intersperses (first push), then switched to `+= '...\n'` which places it at the last entry and needs such hacks.

Not that it really matters, no-one should care about reading `build/platform.c.in` so we could have an empty line either way.

**Review by lumag — APPROVED** [review]

**Pull request merged**